### ISSUE TITLE
How can I make pur work faster?

### ISSUE THREAD

**Issue by avatar-lavventura** [alanhamlett/pip-update-requirements#52]
I observe that updating single packages takes long time than it should be, which is around ~20 seconds.
Is there any way to make this process faster to update only a single package?

```bash
$ time pur -r requirements.txt --only pymysql
Updated pymysql: 1.0.2 -> 1.0.3
All requirements up-to-date.
pur -r requirements.txt --only pymysql  21.13s user 0.65s system 42% cpu 51.593 total
```

**Comment by alanhamlett** [comment]
What's your internet speed? Pur has to talk to Pypi's API to check the latest package versions and there's no way to remove that dependency.

**Comment by avatar-lavventura** [comment]
I feel like when I run `time pur -r requirements.txt --only pymysql` it also checks other packages, which might increase its runtime. Like `pip install -U rich` runs faster than ` pur -r requirements.txt --only rich` on my end. pur is 20 seconds slower.


```bash
# pur
pur -r requirements.txt --only rich  22.19s user 0.83s system 48% cpu 47.622 total

# pip
/home/alper/venv/bin/python3 -m pip install -U rich  2.30s user 0.64s system 89% cpu 3.281 total
```

**Comment by avatar-lavventura** [comment]
Yes sir that was exactly my point, which will speed up the process for a single or few packages.

**Comment by alanhamlett** [comment]
Released in v7.3.0.
